trino/client.py

```diff
@@ -541,7 +544,7 @@ def fetch(self):
     def cancel(self):
         # type: () -> None
         """Cancel the current query"""
-        if self.query_id is None or self.is_finished():
+        if self.query_id is None or self.finished:
             return
 
         self._cancelled = True
```

Comment:
Isn't this prone to incorrectness? The flag should be set AFTER we get a response from co-ordinator.

Pre-existing but worth noting.

Reply:
You mean that `self.query_id` should be set before cancelling?

Reply:
No I'm talking about line 550.

We set the cancelled flag before actually cancelling the query. This can lead to abandoned queries in case the API call to co-ordinator fails for any reason or the client is interrupted.

Reply:
ping @joshthoward . Does what I am saying make sense or did I misunderstand the flow? In case this is an issue we can address this in a follow-up.

Reply:
That does make sense, but let's fix this in a different PR. This one has already been sitting for too long.

Reply:
I agree @joshthoward . Just wanted to make sure we don't forget about this - I'll create an issue to track.

Reply:
@hashhar don't worry about it raising a PR.